Source/EventFlow/Subscribers/DomainEventPublisher.cs

```diff
@@ -101,7 +105,36 @@ public class DomainEventPublisher : IDomainEventPublisher
             CancellationToken cancellationToken)
         {
             var updateReadStoresTasks = _readStoreManagers
-                .Select(rsm => rsm.UpdateReadStoresAsync(domainEvents, cancellationToken));
+                .Select(async rsm =>
+                {
+                    try
+                    {
+                        await _readStoreLog.UpdateBeginAsync(
+                                rsm,
+                                domainEvents,
+                                cancellationToken)
+                            .ConfigureAwait(false);
+                        await rsm.UpdateReadStoresAsync(domainEvents, cancellationToken);
+                    }
+                    catch (Exception e)
+                    {
+                        await _readStoreLog.UpdateFailedAsync(
```

Comment:
May be it is not log? May be this component can have recovery mechanism? An in this case we should not rethrow exception below?
Moreower, exception here will stop processing other side effects - which may be good in some point, and may be not - it should be user decision.
It will lead to missing call `Begin`... for other logs.. So it will lead to more sophisticated recovery process.

Reply:
I think that if any recovery/retry is possible, it should be done within the read store as the read store itself has a knowledge of the underlying technology and its capabilities.

The read store is passed as an argument to the log (maybe not the right name as you said) and you could put any custom additional recovery strategies in there.

Reply:
I mean that if side effect throws exception, `_readStoreLog` (and other logs) can handle it in generic way. 

For example:
```
catch (Exception e)
{
    if (! await _readStoreLog.UpdateFailedAsync(....))
    {
        throw;
    }
```